src/snowflake/snowpark/mock/_connection.py

```diff
@@ -614,9 +614,12 @@ def _fix_pandas_df_integer(table_res: TableEmulator) -> "pandas.DataFrame":
             else:
                 pd_df[pd_df_col_name] = table_res[col_name].astype("int64")
         elif isinstance(col_sf_type.datatype, _IntegralType):
-            pd_df[pd_df_col_name] = pandas.to_numeric(
-                table_res[col_name].tolist(), downcast="integer"
-            )
+            try:
+                pd_df[pd_df_col_name] = table_res[col_name].astype("int64")
```

Comment:
in the mocking, I think we also need to check precision 10 to align with the live behavior

Reply:
The precision check is done when datatype is an instance of `DecimalType`. For `_IntegralType` we don't have precision value. By default, Integral types are NUMBER(38,0) so they fall in `int64` domain.